content/usgs-packages/geoknife_Execute.Rmd

```diff
@@ -0,0 +1,147 @@
+
+```{r other-checks-geojob, message=FALSE, warning=FALSE}
+error(evap_geojob)
+successful(evap_geojob)
+```
+
+The results of all the status checks say that our job was successful!
+
+## Getting geojob data
+
+Since this job has finished processing and was successful, you can now get the data. You'll notice that `evap_geojob` does not actually contain any data. It only contains information about the job that you submitted. To get the data, you need to use `result` or `download`. The stock statistics algorithms will return simple tabular data, so you can use `result` to automatically take the output and parse it into an R `data.frame`. We used a basic stat algorithm in the evapotranspiration example, so let's use `result` to get the `geojob` output.
+
+```{r get-geojob, message=FALSE, warning=FALSE}
+evap_data <- result(evap_geojob)
+nrow(evap_data)
+head(evap_data)
+```
+
+Other stock algorithms could return netcdf or geotiff data. This will require you to handle the output manually using `download`. This will allow you to download the output to a file and then read it using your preferred method (e.g. `read.table`, `read.csv`). See `?download` for more information.
+
+## `wait` and `email`
+
+This was not a computationally or spatially intensive request, so the job finished almost immediately. However, if we had setup a more complex job, it could still be running. Even though the processing of these large gridded datasets uses resources on a remote server, your R code could be impacted by the length of the processes. There are a few scenarios to consider:
+
+1. You are manually executing a job and manually checking it.
+2. You are running a script that kicks off a `geoknife` process followed by lines of code that use the returned data. 
+3. You are running a long `geoknife` process and want to be able to close R/RStudio and come back to a completed job later. 
+
+For the first scenario, the workflow from above was fine. If you are manually checking that the job has completed before trying to extract results, then nothing should fail.
+
+For the second scenario, your code will fail because it will continue to execute the code line by line after starting the job. So, your code will fail at the code that gets the data (`result`/`download`) since the job is still running. You can prevent scripts from continuing until the job is complete by using the function `wait`. This function makes a call to GDP at specified intervals to see if the job is complete, and allows the code to continue once the job is complete. This function has two arguments: the `geojob` object and `sleep.time`. `sleep.time` defines the interval at which to check the status of the job. Please try to adjust `sleep.time` to limit the number of calls to GDP, e.g. if you know the job will take about an hour, set `sleep.time=1800` (a half hour). The default for `sleep.time` is 5 seconds.
```

Comment:
5 minutes seems reasonable.  Even 1 minute would be much better than the default.